flink-table/flink-table-common/src/main/java/org/apache/flink/table/descriptors/CatalogDescriptor.java

```diff
@@ -35,26 +36,49 @@
 
 	private final int propertyVersion;
 
+	private final String defaultDatabase;
+
 	/**
 	 * Constructs a {@link CatalogDescriptor}.
 	 *
 	 * @param type string that identifies this catalog
 	 * @param propertyVersion property version for backwards compatibility
 	 */
 	public CatalogDescriptor(String type, int propertyVersion) {
+		this(type, propertyVersion, null);
+	}
+
+	/**
+	 * Constructs a {@link CatalogDescriptor}.
+	 *
+	 * @param type string that identifies this catalog
+	 * @param propertyVersion property version for backwards compatibility
+	 * @param defaultDatabase default database of the catalog
+	 */
+	public CatalogDescriptor(String type, int propertyVersion, String defaultDatabase) {
 		this.type = type;
 		this.propertyVersion = propertyVersion;
+		this.defaultDatabase = defaultDatabase;
 	}
 
 	@Override
 	public final Map<String, String> toProperties() {
 		final DescriptorProperties properties = new DescriptorProperties();
 		properties.putString(CATALOG_TYPE, type);
 		properties.putLong(CATALOG_PROPERTY_VERSION, propertyVersion);
+
+		if (defaultDatabase != null) {
+			properties.putString(CATALOG_DEFAULT_DATABASE, defaultDatabase);
+		}
+
 		properties.putProperties(toCatalogProperties());
 		return properties.asMap();
 	}
 
+	public String getDefaultDatabase() {
+		return defaultDatabase;
+	}
+
 	/**
 	 * Converts this descriptor into a set of catalog properties.
 	 */
```

Comment:
Side q: what's the purpose of this method?

Reply:
this pattern is borrowed from existing descriptors. `toProperties()` is for converting common configs for all catalogs, and `toCatalogProperties()` is for converting catalog-specific configs. Impl of `toCatalogProperties()` will always call `super.toProperties()` and then run their own own logic.